### ISSUE TITLE
When open a link from another app, ublock doesn't work

### ISSUE THREAD

**Issue by superlex** [el1t/uBlock-Safari#58]
Description:
When open a link from another app, ublock doesn't work. It occurs if your default browser is Safari: if your default browser is e.g. Firefox + ublock-origin, it doesn't occur.

Steps to reproduce:
- Set Safari as your default browser;
- Close Safari if it's open;
- Paste a link of a site that has ads in another app, e.g. Notes;
- Click on this link: Safari will open, but ads are still there;
- Reload the page or paste the link directly in Safari: ublock works fine and ads go away.

Info:
- MacOS version: 10.12.6
- Safari version: 10.1.2
- uBlock Origin version: 1.12.5.100

**Comment by superlex** [comment]
uBlock Origin 1.13.11b doesn't fix the issue.

**Comment by ServerWrestler** [comment]
@superlex I was not able to reproduce with 1.13.11.100.  I wonder if anybody else is experiencing the issue?

**Comment by superlex** [comment]
@the-sillystring 
Thanks for reply.
Did you try closing safari with cmd+q?

**Comment by ServerWrestler** [comment]
@superlex Yep I closed with cmd+q.  Then typed https://www.yahoo.com into Notes and clicked on it.  Over 30 items were blocked.  Odd.  Do you have any other extensions installed?

**Comment by superlex** [comment]
Nope, only uBlock Origin.
Could you try with https://www.apkmirror.com ?
Counter says 13 items were blocked, but they are there. When I reload the page ads disappear, however counter says 13 again.

**Comment by ServerWrestler** [comment]
I just tested with https://www.apkmirror.com and it worked fine.  13 items were blocked.  Unfortunately I'm unable to reproduce the issue.  Doesn't mean the bug doesn't exist though.

**Comment by superlex** [comment]
Maybe it depends on filter lists.
I've a lot of lists enabled, I tried to disable someone and I noticed that in this way sometimes I can reproduce this bug but sometimes not.

**Comment by gorhill** [comment]
> Counter says 13 items were blocked, but they are there.

The counter reports blocked network requests. The elements you see may be due to cosmetic filtering not being applied -- cosmetically filtered elements are not reported in the counter.

**Comment by superlex** [comment]
@gorhill 
Thanks for partecipating.
Yes, you're right. In fact enabling/disabling cosmetic filtering (with the "eye" icon) has no effect. Only reloading the page it works properly.

Info:
MacOS version: 10.12.6
Safari version: 11.0.2
uBlock Origin version: 1.14.14
